### PR TITLE
Added listener options and dialer options

### PR DESCRIPTION
**Pull request by wanderer**

### CODE REVIEW

File: tests/base-test.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 var multiaddr = require('multiaddr')
 
-module.exports.all = function (test, common) {
+module.exports.all = function (test, common, listenerOpts, dialerOpts) {
```
Comment (diasdavid) [inline tests/base-test.js]: let's add a check default to `{}` if none are passed

**Comment by diasdavid** [comment]
yes, like this, thank you :)

File: tests/base-test.js (hunk 1 above)
Comment (wanderer) [inline tests/base-test.js]: @diasdavid done!

**Comment by wanderer** [comment]
@diasdavid ok I updated the PR. I also changed the location of where the option a passed to the setup method. This is because I want to use the same format for simulation where the setup method will be called multiple times.

**Pull request closed without merging**